downstairs/src/dump.rs

```diff
@@ -353,14 +394,34 @@ fn show_extent(
             }
         }
 
-        print!("{0:^11} {1:^11} ", status_letters[0], status_letters[1],);
-        if dir_count > 2 {
-            print!("{0:^11} ", status_letters[2]);
+        for dir_index in 0..dir_count {
+            tag_columns[dir_index] =
+                format!("{0:^11} ", status_letters[dir_index]);
+        }
+
+        if !only_show_differences || different {
+            print!("Block {:4}", block);
+
+            for column in data_columns.iter().take(dir_count) {
+                print!("{}", column);
+            }
+            for column in nonce_columns.iter().take(dir_count) {
+                print!("{}", column);
+            }
+            for column in tag_columns.iter().take(dir_count) {
+                print!("{}", column);
+            }
+
+            print!("{0:^7}", if different { "<-------" } else { "" });
```

Comment:
Nit: If you are only printing differences, we could probably skip the DIFF column all together.